### DISCUSSION TITLE
cnabquickstarts.azurecr.io/cnabarmdriver Image Documentation

### DISCUSSION THREAD

**Discussion by christianrowlands** [getporter/porter#1829] [Q&A] [answered]
Hello, I am working on creating an Azure Marketplace listing that will kick off a CNAB installer. Thanks to @carolynvs , I was able to get past [this issue](https://github.com/getporter/porter/issues/1827) which was the result of a stupid mistake I made.

Since we are kicking off a CNAB installer from ACI, I have found the [azure-cnab-quickstarts](https://github.com/Azure/azure-cnab-quickstarts) repo to be helpful. It seems that the approach taken in these quick start examples point to the `cnabquickstarts.azurecr.io/cnabarmdriver:0.25.0` image in [Azure Container Registry](https://github.com/Azure/azure-cnab-quickstarts/blob/main/porter/aks/azuredeploy-simple.json#L128) as the bootstrap point to then kick off another ACI that runs the actual [porter CNAB container](https://github.com/Azure/azure-cnab-quickstarts/blob/main/porter/aks/azuredeploy-simple.json#L186).
 
That flow makes sense to me, but I am unable to find any documentation on the cnabarmdriver container. The reason I am looking for documentation is because I am unsure how to point that bootstrap container at my custom CNAB bundle that will likely be hosted outside of Azure Container Registry or a non public repo (I am currently using a public Docker Hub repo and it is working). More specifically, let’s say I host my CNAB porter container in my company's Harbor registry that is not public. How would I pass in the Harbor registry credentials so that the container can be pulled down? In Kubernetes I can set the imagePullSecrets value in the deployment spec so that the Kubernetes cluster can authenticate against Harbor (or any container registry). Is there something like that for the cnabarmdriver image?

To summarize, I am am looking for any documentation on the `cnabquickstarts.azurecr.io/cnabarmdriver` image, or if that is not the right image to use, is there another driver/bootstrap image that I should be leveraging as the "middleware glue" between the Marketplace Deployment Template and my porter CNAB installer?

I appreciate the help!

- Christian

**Comment by simongdavies** [comment, marked as answer]
@christianrowlands please take a look at https://github.com/deislabs/cnab-azure-driver . I believe that what you need to do is to set the environment variables `CNAB_AZURE_REGISTRY_USERNAME` and `CNAB_AZURE_REGISTRY_PASSWORD`. Let me know if this isn't what you need.

**Reply by christianrowlands** [reply]
That might be what I need. I saw _"The main purpose of this driver is to enable CNAB operations to be performed using Azure CloudShell."_ in the description of that repo and thought it was for something else, but I think I assumed too much there.

I will give it a try and see if it works.

Thanks!

**Reply by simongdavies** [reply]
> The main purpose of this driver is to enable CNAB operations to be performed using Azure CloudShell

Oh, thanks, I updated the text to make it a little clearer that its not the only purpose.
